Automattic/wp-calypso
# [Private media files] Use API proxy for private atomic sites

#### Changes proposed in this Pull Request

This PR is related to [private atomic sites project](https://wp.me/paObgF-Ui). It makes it possible for Calypso to load images from remote private site by using public-api API endpoint instead of requesting the image directly. The changes include:

* Data layer which specifies its `responseType` is a `blob`
* `ProxiedImage` component which calls the above & uses URL.createObjectURL to set the image to the response Blob
* Use `ProxiedImage` when a site is Atomic & Private

#### Testing instructions

* Apply D39236-code (proxy provider script)
* Sandbox the REST API
* Visit `/media` for a Private Atomic site, confirm all images are displayed correctly. Note the loading is going to take some time as we don't support thumbnails yet.
* Visit `/media` for a a regular non-private Atomic site, confirm all images are displayed correctly.

#### Notes

* This PR doesn't address specific images preview/editing just yet
